sailthru_content/sailthru_content.py

```diff
@@ -0,0 +1,317 @@
+#! /usr/bin/env python
+
+""" Synchronize edX course catalog in to Sailthru. """
+
+import logging
+import datetime
+import json
+import sys
+import os
+import argparse
+from edx_rest_api_client.client import EdxRestApiClient
+from sailthru.sailthru_client import SailthruClient
+
+logger = logging.getLogger(__name__)
+
+
+def setup_logging():
+    """ Setup logging.
+
+    Add support for console logging, and avoid truncation by pandas.
+    """
+    msg_format = '%(asctime)s - %(levelname)s - %(message)s'
+    ch = logging.StreamHandler()
+    ch.setFormatter(logging.Formatter(msg_format))
+    ch.setLevel(logging.DEBUG)
+    logger.addHandler(ch)
+    logger.setLevel(logging.DEBUG)
+
+
+def main(argv):
+    parser = argparse.ArgumentParser(description='Sailthru course synch script')
+
+    parser.add_argument(
+            'command',
+            choices=['list', 'load', 'clear'])
+
+    parser.add_argument(
+            '--access_token',
+            default=os.environ.get('CONTENT_LOAD_ACCESS_TOKEN', None),
+            help='OAuth2 access token used to authenticate API calls.'
+        )
+
+    parser.add_argument(
+            '--oauth_host',
+            default=os.environ.get('CONTENT_LOAD_OAUTH_HOST', 'https://courses.stage.edx.org/oauth2'),
+            help='OAuth2 base url.'
+        )
+
+    parser.add_argument(
+            '--oauth_key',
+            default=os.environ.get('CONTENT_LOAD_OAUTH_KEY', None),
+            help='OAuth2 key used to authenticate Course Content API calls.'
+        )
+
+    parser.add_argument(
+            '--oauth_secret',
+            default=os.environ.get('CONTENT_LOAD_OAUTH_SECRET', None),
+            help='OAuth2 secret token used to authenticate Course Content API calls.'
+        )
+
+    parser.add_argument(
+            '--sailthru_key',
+            default=os.environ.get('CONTENT_LOAD_SAILTHRU_KEY', None),
+            help='Sailthru access key.'
+        )
+
+    parser.add_argument(
+            '--sailthru_secret',
+            default=os.environ.get('CONTENT_LOAD_SAILTHRU_SECRET', None),
+            help='Sailthru access secret.'
+        )
+
+    parser.add_argument(
+            '--content_api_url',
+            default=os.environ.get('CONTENT_LOAD_API_URL', None),
+            help='Content api url.'
+        )
+
+    args = parser.parse_args()
+    access_token = args.access_token
+
+    # connect to Sailthru
+    sc = SailthruClient(args.sailthru_key, args.sailthru_secret)
+
+    # Process 'list' command
+    if args.command == 'list':
+        while True:
+            response = sc.api_get('content', {'items': 1000})
+
+            if not response.is_ok():
+                logger.error("Error code %d connecting to Sailthru content api: %s",
+                             response.json['error'], response.json['errormsg'])
+                return
+
+            for body in response.json['content']:
+                logger.info(body)
+            return
```

Comment:
This `return` means that you only run through the `while True` loop once. Is that just a temporary debugging measure?